cube/src/main/java/zone/cogni/asquare/cube/index/ElasticsearchIndexingService.java

```diff
@@ -0,0 +1,604 @@
+package zone.cogni.asquare.cube.index;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.vavr.control.Try;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.rdf.model.RDFNode;
+import org.elasticsearch.index.IndexNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+import zone.cogni.asquare.cube.convertor.ModelToJsonConversion;
+import zone.cogni.asquare.cube.monitoredpool.MonitoredPool;
+import zone.cogni.asquare.cube.pagination.PaginatedQuery;
+import zone.cogni.asquare.cube.sparql2json.SparqlSelectToJson;
+import zone.cogni.asquare.cube.spel.SpelService;
+import zone.cogni.asquare.service.elasticsearch.v7.Elasticsearch7Store;
+import zone.cogni.asquare.triplestore.RdfStoreService;
+
+import javax.annotation.Nonnull;
+import javax.annotation.PostConstruct;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Here is an expected folder structure for indexing:
+ *
+ * <pre>
+ *   /rootFolder                <- root folder
+ *     /name-1                  <- name of index
```

Comment:
future update: have a mapping per environment, so we can have same folder structure but a different index name per environment (or per installation)

Reply:
see issue #41